### PR TITLE
Documentation: Sun Module + Libraries

### PR DESCRIPTION
**Pull request by silochad**

### CODE REVIEW

**Review by Brean0 — CHANGES_REQUESTED** [review]

File: protocol/contracts/C.sol [hunk 1]
```diff
@@ -91,10 +90,6 @@ library C {
     // Use external contract for block.basefee as to avoid upgrading existing contracts to solidity v8
     address private constant BASE_FEE_CONTRACT = 0x84292919cB64b590C0131550483707E43Ef223aC;
 
-    /**
-     * Getters
```
[the remark this answers is not in the document]
Comment (silochad) [reply protocol/contracts/C.sol:95]: Going to refactor the whole file later, probably a larger change than we have time for right now

File: protocol/contracts/beanstalk/sun/SeasonFacet/SeasonFacet.sol [hunk 2]
```diff
@@ -10,86 +10,129 @@ import "~/libraries/LibIncentive.sol";
...
+
+    //////////////////// SUNRISE ////////////////////
 
-    /// @notice advances Beanstalk to the next Season, sending reward beans to the caller's circulating balance
-    /// @return reward The number of beans minted for the caller.
+    /**
+     * @notice Advances Beanstalk to the next Season, sending reward Beans to the caller's circulating balance.
+     * @return reward The number of beans minted to the caller.
+     */
     function sunrise() external payable returns (uint256) {
```
[the remark this answers is not in the document]
Comment (silochad) [reply protocol/contracts/beanstalk/sun/SeasonFacet/SeasonFacet.sol:42]: Shouldn't the _addition_ of a return value cause no backwards incompatibility issue? I.e. anyone who read this previously just ignored the return, and the return doesn't affect the function signature ABI at all.

File: protocol/contracts/beanstalk/sun/SeasonFacet/SeasonFacet.sol [hunk 3]
```diff
@@ -10,86 +10,129 @@ import "~/libraries/LibIncentive.sol";
...
 
+    /**
+     * @param account The address to which the reward beans are sent, may or may not
+     * be the same as the caller of `sunrise()`
+     * @param initialGasLeft The amount of gas left at the start of the transaction
+     * @param balances The current balances of the BEAN:3CRV pool returned by {stepOracle}
+     * @param mode Send reward beans to Internal or Circulating balance
+     * @dev Mints Beans to `account` as a reward for calling {sunrise()}.
+     */
     function incentivize(
```
[the remark this answers is not in the document]
Comment (silochad) [reply protocol/contracts/beanstalk/sun/SeasonFacet/SeasonFacet.sol:136]: Defer to your opinion and @publiuss' here, but personally would lean towards simplicity in calculation for the general case and letting this case happen occasionally, if the extent of the problem would be that a `sunrise()` caller can receive an extra 1 block's worth of reward.

File: protocol/contracts/beanstalk/sun/SeasonFacet/SeasonFacet.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (Brean0) [reply protocol/contracts/beanstalk/sun/SeasonFacet/SeasonFacet.sol:42]: makes sense, https://solidity-by-example.org/function-selector/ does not include return values

**Review by Brean0 — APPROVED** [review]

**Review by hellofromguy — APPROVED** [review]

**Review by Brean0 — APPROVED** [review]

**Pull request merged**